internal/infrastructure/middleware/enforce_identity_test.go

```diff
@@ -578,3 +567,99 @@ func TestNewEnforceOr(t *testing.T) {
 		}
 	}
 }
+
+func TestEnforceServiceAccountPredicate(t *testing.T) {
+	type TestCase struct {
+		Name     string
+		Given    *identity.XRHID
+		Expected error
+	}
+	testCases := []TestCase{
+		{
+			Name:     "nil argument",
+			Given:    nil,
+			Expected: fmt.Errorf("'data' cannot be nil"),
+		},
+		{
+			Name: "'Identity.Type' is not 'ServiceAccount'",
+			Given: &identity.XRHID{
+				Identity: identity.Identity{
+					Type: "System",
+				},
+			},
+			Expected: fmt.Errorf("'Identity.Type' must be 'ServiceAccount'"),
+		},
+		{
+			Name: "'Identity.AuthType' must be 'jwt-auth'",
+			Given: &identity.XRHID{
+				Identity: identity.Identity{
+					Type:     "ServiceAccount",
+					AuthType: "cert",
+				},
+			},
+			Expected: fmt.Errorf("'Identity.AuthType' is not 'jwt-auth'"),
+		},
+		{
+			Name: "'Identity.ServiceAccount' is nil",
+			Given: &identity.XRHID{
+				Identity: identity.Identity{
+					Type:           "ServiceAccount",
+					AuthType:       "jwt-auth",
+					ServiceAccount: nil,
+				},
+			},
+			Expected: fmt.Errorf("'Identity.ServiceAccount' is nil"),
+		},
+		{
+			Name: "'Identity.ServiceAccount.ClientId' is empty",
+			Given: &identity.XRHID{
+				Identity: identity.Identity{
+					Type:     "ServiceAccount",
+					AuthType: "jwt-auth",
+					ServiceAccount: &identity.ServiceAccount{
+						ClientId: "",
+					},
+				},
+			},
+			Expected: fmt.Errorf("'Identity.ServiceAccount.ClientId' is empty"),
+		},
+		{
+			Name: "'Identity.ServiceAccount.ClientId' is empty",
```

Comment:
The test name should be `'Identity.ServiceAccount.Username' is empty`

Reply:
Thanks! fixed!